ui/apps/dashboard/src/app/(auth)/sign-up/account-setup/page.tsx

```diff
@@ -68,6 +59,74 @@ export default function AccountSetupPage() {
   );
 }
 
+/**
+ * This forces the user to have an active organization when the organizations feature is not
+ * enabled. When the organizations feature is enabled, the user will already have an active
+ * organization set up by the `/organization-list` page.
+ */
+function useForceActiveOrganization() {
+  const { value: isOrganizationsEnabled } = useBooleanFlag('organizations');
+  const { isLoaded, userMemberships, setActive } = useOrganizationList({
+    userMemberships: {
+      infinite: true,
+    },
+  });
+  const { organization } = useOrganization();
+
+  useEffect(() => {
+    if (!isOrganizationsEnabled && isLoaded && userMemberships.data[0] && !organization) {
+      setActive({ organization: userMemberships.data[0].organization });
+    }
+  }, [isLoaded, isOrganizationsEnabled, organization, setActive, userMemberships.data]);
+
+  // Reload memberships if the user doesn't have an active organization
+  useEffect(() => {
+    if (isOrganizationsEnabled || !isLoaded || userMemberships.data[0] || organization) return;
```

Comment:
Should this have a negation?
```suggestion
    if (isOrganizationsEnabled || !isLoaded || !userMemberships.data[0] || organization) return;
```
Would be good to roll this up into a boolean with a `useMemo()` or something which these `useEffect()` hooks then call, as it's hard to tell what we're checking with so many conditions being listed.

Reply:
It shouldn't be a negation. We want to return early if a user membership is already present. We don't want to keep reloading memberships if we already have one.